server/sonar-server/src/main/java/org/sonar/server/organization/ws/UpdateProjectVisibilityAction.java

```diff
@@ -62,20 +68,29 @@ public void define(WebService.NewController context) {
     action.createParam(PARAM_PROJECT_VISIBILITY)
       .setRequired(true)
       .setDescription("Default visibility for projects")
-      .setPossibleValues("private", "public");
+      .setPossibleValues(PRIVATE_PROJECT_VISIBILITY, PUBLIC_PROJECT_VISIBILITY);
   }
 
   @Override
   public void handle(Request request, Response response) throws Exception {
     String organizationKey = request.mandatoryParam(PARAM_ORGANIZATION);
-    boolean newProjectsPrivate = "private".equals(request.mandatoryParam(PARAM_PROJECT_VISIBILITY));
+    boolean newProjectsPrivate = PRIVATE_PROJECT_VISIBILITY.equals(request.mandatoryParam(PARAM_PROJECT_VISIBILITY));
     try (DbSession dbSession = dbClient.openSession(false)) {
       Optional<OrganizationDto> optionalOrganization = dbClient.organizationDao().selectByKey(dbSession, organizationKey);
       OrganizationDto organization = checkFoundWithOptional(optionalOrganization, "No organization with key '" + organizationKey + "' can be found.");
       userSession.checkPermission(OrganizationPermission.ADMINISTER, organization.getUuid());
+      checkCanUpdateProjectsVisibility(organization, newProjectsPrivate);
       dbClient.organizationDao().setNewProjectPrivate(dbSession, organization, newProjectsPrivate);
       dbSession.commit();
     }
     response.noContent();
   }
+
+  private void checkCanUpdateProjectsVisibility(OrganizationDto organization, boolean newProjectsPrivate) {
+    try {
+      billingValidations.checkCanUpdateProjectsVisibility(new BillingValidations.Organization(organization.getKey(), organization.getUuid()), newProjectsPrivate);
+    } catch (BillingValidations.BillingValidationsException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)